### PR TITLE
Change formats source

### PR DESCRIPTION
**Pull request by RocoGreen**
I know i already do a pull request but it was a problem with progress event and i fixed it.

### CODE REVIEW

**Comment by RocoGreen** [comment]
This is the last edit but if there are a bug, i will fix it

File: lib/info.js [hunk 1]
```diff
@@ -97,16 +97,18 @@ exports.getBasicInfo = (id, options, callback) => {
  * @return {Array.<Object>}
  */
 const parseFormats = (info) => {
-  let formats = [];
-  if (info.url_encoded_fmt_stream_map) {
-    formats = formats.concat(info.url_encoded_fmt_stream_map.split(','));
+  let formats = []
+
+  if (info.player_response.streamingData.formats) {
```
Comment (fent) [inline lib/info.js]: is `streamingData` always defined? thinking that it should be checked if it's not

File: lib/info.js [hunk 2]
```diff
@@ -97,16 +97,18 @@ exports.getBasicInfo = (id, options, callback) => {
...
+  formats = formats.concat(info.player_response.streamingData.formats)
```
Comment (fent) [inline lib/info.js]: should be indented

File: lib/info.js [hunk 3]
```diff
@@ -97,16 +97,18 @@ exports.getBasicInfo = (id, options, callback) => {
...
   }
-  if (info.adaptive_fmts) {
-    formats = formats.concat(info.adaptive_fmts.split(','));
+  if (info.player_response.streamingData.adaptiveFormats) {
+  formats = formats.concat(info.player_response.streamingData.adaptiveFormats)
```
Comment (fent) [inline lib/info.js]: same comments here as above

File: lib/util.js [hunk 4]
```diff
@@ -34,7 +34,7 @@ exports.sortFormats = (a, b) => {
 
   const getBitrate = (c) => {
     if (c.bitrate) {
-      let s = c.bitrate.split('-');
+      let s = c.bitrate.toString().split('-');
```
Comment (fent) [inline lib/util.js]: is this a number now? if so, it wouldn't need to be `.split()`d anymore

Comment (RocoGreen) [reply lib/util.js]: Sometimes, it's a number and sometimes not that's why i added `.toString()`

File: lib/info.js (hunk 2 above)
Comment (RocoGreen) [reply lib/info.js]: Ok, fixed

File: lib/info.js (hunk 3 above)
Comment (RocoGreen) [reply lib/info.js]: Fixed too

**Comment by SwissCheese5** [comment]
Any updates on this?

**Comment by fent** [comment]
merged in https://github.com/fent/node-ytdl-core/commit/eb1491f5f70302a2b11005041aa9cd0f4a53ef08

**Pull request closed without merging**